CHANGELOG.md

```diff
@@ -69,6 +69,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Fix 'org.apache.hc.core5.http.ParseException: Invalid protocol version' under JDK 16+ ([#4827](https://github.com/opensearch-project/OpenSearch/pull/4827))
 - Fixed compression support for h2c protocol ([#4944](https://github.com/opensearch-project/OpenSearch/pull/4944))
 - Support OpenSSL Provider with default Netty allocator ([#5460](https://github.com/opensearch-project/OpenSearch/pull/5460))
+- Fix weighted shard routing state across search requests([#6004](https://github.com/opensearch-project/OpenSearch/pull/6004))
```

Comment:
Do you intend to backport this to 2.x? If so, you'll need to [put this in the `[Unreleased 2.x]` section](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#where-should-i-put-my-changelog-entry).

Reply:
yes, moved the entry to 2.x section. Thanks for pointing this out.

Reply:
> > This PR fixes issue due to which state is not maintained across weighted shard routing search requests .
> 
> Can you provide more detail about what is changing here (or link an issue that explains in detail what the problem is)? "state is not maintained" is the cause, but what is the actual issue? i.e. what is the behavior change that a user will as a result of this fix?

Created an issue with the details https://github.com/opensearch-project/OpenSearch/issues/6056